src/Service/ClientServiceFactory.php

```diff
@@ -14,7 +14,11 @@ class ClientServiceFactory
     public function __invoke($container)
     {
         $config     = $container->get('config');
-        $httpClient = new HttpClient();
+        $httpClient = new HttpClient(
+            null,
+            !empty($config['apigility-consumer']['api-http-adapter-options']) ?
```

Comment:
its fine, im gonna work on my own fork and continue there, obvious the perspectives divege